olive/passes/onnx/perf_tuning.py

```diff
@@ -313,24 +290,37 @@ def __init__(self, accelerator_spec: AcceleratorSpec, config: Dict[str, Any], da
         self.data_root = data_root
 
     def tune_onnx_model(self, model):
-        latency_user_config = {}
-        # which should be the same as the config in the metric
-        config_dict = self.config.dict()
-
-        # data_dir/dataloader_func will be passed to the metric as perf_tuning will leverage
-        # the latency metric to run tune
-        for eval_config in get_user_config_properties_from_metric_type(MetricType.LATENCY):
-            if eval_config in config_dict:
-                latency_user_config[eval_config] = config_dict.get(eval_config)
-        if config_dict.get("dataloader_func_kwargs"):
-            latency_user_config["func_kwargs"] = {"dataloader_func": config_dict.get("dataloader_func_kwargs")}
+        if not self.config.data_config:
+            if model.io_config:
+                if not is_io_config_static(model.io_config):
+                    # since Olive will not save the pytorch model's io_config to olive onnx model
+                    # we cannot generate dummy data for the onnx model if this model has dynamic input shapes
+                    # TODO(trajep): try to get static input shapes from onnx model.
+                    # If so, we can move the dataloader for latency measurement.
+                    logger.debug(
```

Comment:
should we make this a failure since this case is bound to fail?

Reply:
Keeping the same behavior as earlier. Handling of `io_config` should be its own PR.